cocotb/share/lib/vpi/VpiImpl.cpp

```diff
@@ -547,18 +547,9 @@ int VpiImpl::deregister_callback(GpiCbHdl *gpi_hdl)
     return 0;
 }
 
-// If the Python world wants things to shut down then unregister
-// the callback for end of sim
 void VpiImpl::sim_end()
 {
-    /* Some sims do not seem to be able to deregister the end of sim callback
-     * so we need to make sure we have tracked this and not call the handler
-     */
-    if (GPI_DELETE != sim_finish_cb->get_call_state()) {
-        sim_finish_cb->set_call_state(GPI_DELETE);
-        vpi_control(vpiFinish);
```

Comment:
Was this not needed at all?

Reply:
The first bit is deregistering the shutdown callback. Seemingly to prevent it from running because it would try to run through shutdown behavior again once the simulator actually shuts down. That should certainly be tossed, that's part of the normalization.

`vpiFinish` is important, I just though that through this morning, I will add that back in.